### ISSUE TITLE
Windows: cc_binary has multiple output artifacts when depending on DLLs

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#8707]
```
cc_binary(
  name = "foo.dll",
  srcs = ["foo.cc"],
)
```
In a normal case, building `//foo.dll` will only generate one output artifact `foo.dll` in the default output group.

When the cc_binary depends on an dll, for example:
```
cc_import(
  name = "bar",
  shared_library = "bar.dll",
  interface_library = "bar.lib",
)

cc_binary(
  name = "foo.dll",
  srcs = ["foo.cc"],
  deps = ["//:bar"],
)
```
In this case, Bazel uses `bar.lib` at `foo.dll`'s linking time and copies `bar.dll` to the directory of `foo.dll` in order to make it available at runtime. But, Bazel also adds `bar.dll` to the default output group of `foo.dll`, which is problematic because when users try to reference `foo.dll` via `//bar:foo.dll`, it actually have more than one output artifact. This could be very confusing and there is no good way to distinguish the main output (`foo.dll`) from its dll dependencies (`bar.dll`).

One way to solve this is to add all DLL dependencies to a separate output group (eg. `runtime_dynamic_libraries`) and make sure the default output group of a cc_binary will always have a single artifact.

**Comment by meteorcloudy** [comment]
@baskus @brills

**Comment by baskus** [comment]
@meteorcloudy Great! Thanks for fixing this!

What release do you expect this fix will make it into?

**Comment by meteorcloudy** [comment]
This will be in next month's release (0.28.0) ;)

**Comment by meteorcloudy** [comment]
@baskus @siberex 2260b970cb2a10c21a2fde116aa82c79be52991f fixed a bug caused by 210428c , if you're cherry-picking 210428c, you should also need this fix.

**Comment by meteorcloudy** [comment]
FYI, 0.28.0 has 210428c but not the fix, I'm asking for a cherry-pick into 0.28.1.
